main.c

```diff
@@ -0,0 +1,124 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+struct Node {
+  int data;
+  struct Node* nextEntry;
+  struct Node* prevEntry;
+};
+
+struct Node* head;
+
+struct Node* CreateNode (int var)
+{
+  struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
```

Comment:
Welcome to C world, if malloc is used you have to make sure that it managed to allocate the memory, as well that it was ```free()```d after it's no longer needed. Memory-leaks C style.